### ISSUE TITLE
Run e2etest for back-restore tests suite in parallel

### ISSUE THREAD

**Issue by sayanh** [kyma-project/kyma#4158]
**Description**

Since it is sequential, one failing test breaks the whole run of running tests. 
Running it in parallel will not only reduce time to execute the suite but better debugging abilites.

May be goconvey is not the right tool here. We could use octopus with pure Golang execution with goroutines.

**Comment by clebs** [comment]
This would be a nice to have improvement, but it requires a complete testing framework change and it is not feasible to do any time soon.

We might revisit this in the future.
